src/js/media-controller.js

```diff
@@ -0,0 +1,309 @@
+  * Take custom controls to fullscreen
+  * Position controls at the bottom
+  * Auto-hide controls on inactivity while playing
+*/
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import { propagateMedia, setAndPropagateMedia } from './media-chrome-html-element.js';
+import { Window as window, Document as document } from './utils/server-safe-globals.js';
+import { MEDIA_PLAY_REQUEST, MEDIA_PAUSE_REQUEST } from './media-ui-events.js';
+
+const template = document.createElement('template');
+
+template.innerHTML = `
+  <style>
+    :host {
+      box-sizing: border-box;
+      position: relative;
+
+      /* Position controls at the bottom  */
+      display: flex;
+      flex-direction: column-reverse;
+
+      /* Default dimensions */
+      width: 100%;
+      max-width: 720px;
+      height: 480px;
+      background-color: #000;
+    }
+
+    /* Safari needs this to actually make the element fill the window */
+    :host(:-webkit-full-screen) {
+      /* Needs to use !important otherwise easy to break */
+      width: 100% !important;
+      height: 100% !important;
+    }
+
+    /* Position the media element to fill the container */
+    ::slotted([slot=media]) {
+      position: absolute;
+      top: 0;
+      left: 0;
+      width: 100%;
+      height: 100%;
+    }
+
+    /* Hide controls when inactive and not paused */
+    #container ::slotted(*) {
+      opacity: 1;
+      transition: opacity 0.25s;
+      visibility: visible;
+    }
+
+    #container.inactive:not(.paused) ::slotted(*) {
+      opacity: 0;
+      transition: opacity 1s;
+    }
+  </style>
+  <slot name="media"></slot>
+  <div id="container">
+    <slot></slot>
+  </div>
+`;
+
+/*
+  Loop through child nodes and set the media[State] on every child.
+*/
+function propagateMediaState(el, stateName, val) {
+  Array.prototype.forEach.call(el.children, (child) => {
+    const childName = child.nodeName.toLowerCase();
+
+    if (child.slot == 'media') return;
+
+    function setAndPropagate() {
+      // Only set if previously defined, at least as null
+      // This is how element authors can tell us they want to
+      // receive these state updates
+      if (typeof child[stateName] !== 'undefined') {
+        child[stateName] = val;
+      }
+      propagateMediaState(child, stateName, val);
+    }
+
+    // Make sure custom els are ready
+    if (childName.includes('-') && !window.customElements.get(childName)) {
+      window.customElements.whenDefined(childName).then(setAndPropagate);
+    } else {
+      setAndPropagate();
+    }
+  });
+}
+
+class MediaController extends window.HTMLElement {
+  constructor() {
+    super();
+
+    // Set up the Shadow DOM
+    const shadow = this.attachShadow({ mode: 'open' });
+    this.shadowRoot.appendChild(template.content.cloneNode(true));
+    this.container = this.shadowRoot.getElementById('container');
+
+    // Watch for child adds/removes and update the media element if necessary
```

Comment:
This is mostly carried over from media-container, but handles any changing of the contained media element.